### PR TITLE
Twitter Inference Blueprint: Adding Twitter Connector

### PR DESCRIPTION
**Pull request by yidinghabana**
Adding a new folder for ft-twitter-connector to this blueprint. The twitter connector was available in Dev and is re-used.  

pull_twitter_data.py
library.yaml
README.md
requirements.txt

### CODE REVIEW

File: ft-twitter-connector/pull_twitter_data.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+
+class EndPointAction(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        if not values == 'recent' and not values == 'all':
+            raise argparse.ArgumentError(self, "endpoint must be 'recent'  or 'all' ")
+        setattr(namespace, self.dest, values)
+
+
+parser = argparse.ArgumentParser(description="""Preprocessor""")
```
Comment (orionsBeltWest) [inline ft-twitter-connector/pull_twitter_data.py]: Try to not have any global code loose. Please go ahead and put arguparse code under a function object.

**Review by orionsBeltWest — CHANGES_REQUESTED** [review]
In this pull request I see only one python file is changed!  pull_twitter_data.py : It has loose global code. 

Also in this repo I do see several issues in files not changed:
1) I see only one unit test in the ft-fine-tuning-inference section.  No other unittest are present. 
2) global code reading yaml is in ft-fine-tuning-inference-twitter/fine-tune-inference.py
3) same global issue in ft-fine-tune/fine-tune.py
4) not sure I spotted all the global code every place.   But try to fix when every possible. 
5) recommend running pylint on code.

File: ft-twitter-connector/pull_twitter_data.py (hunk 1 above)
Comment (yidinghabana) [reply ft-twitter-connector/pull_twitter_data.py]: I refactorized the old twitter code to submit for review by following our current code requirement.

Comment (orionsBeltWest) [reply ft-twitter-connector/pull_twitter_data.py]: code refactoring looks good

File: ft-fine-tune/fine_tune_unittests.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+    def test_list_values(self):
+        """Checks if function returns list with just '2, 1, 0's"""
+        returned_lists_trainy = self.trainy.apply(lambda x: decode_sentiment(x))
+        # print("returned_lists_trainy", returned_lists_trainy)
+        returned_lists_validy = self.validy.apply(lambda x: decode_sentiment(x))
+        # print("returned_lists_validy", returned_lists_validy)
+        self.assertEqual(all(returned_lists_trainy), all(self.train_mapped))
+        self.assertEqual(all(returned_lists_validy), all(self.valid_mapped))
+
```
Comment (orionsBeltWest) [inline ft-fine-tune/fine_tune_unittests.py:132]: fine_tune_unittests.py missing main:
```python:
if __name__ == "__main__":
    unittest.main()
```

**Review by orionsBeltWest — CHANGES_REQUESTED** [review]
The code refactoring is okay. 

Just the unittest file need main:

File: ft-fine-tune/fine_tune_unittests.py (hunk 2 above)
Comment (yidinghabana) [reply ft-fine-tune/fine_tune_unittests.py:132]: Thanks Basem for your review. 

I tested these two lines to add into unittest.py or not included into unittest.py, both ways work. Per our discussion, I will include them in all the future unittest.py.

Comment (orionsBeltWest) [reply ft-fine-tune/fine_tune_unittests.py:132]: Kindly please add it in this pull request although things seem to run without it. 

unittest.main() define the default arguments:

``` python:
unittest.main(module='__main__', defaultTest=None, argv=None, testRunner=None, testLoader=unittest.defaultTestLoader, exit=True, verbosity=1, failfast=None, catchbreak=None, buffer=None, warnings=None)
```

Comment (yidinghabana) [reply ft-fine-tune/fine_tune_unittests.py:132]: Sure, I will add it and re_submit it for approval!

**Review by orionsBeltWest — APPROVED** [review]

**Review by Ramji-Satya — APPROVED** [review]

**Pull request merged**